File: leetcode/3-Palindrome-Number/palindrome-number.go
//Runtime: 48 ms, faster than 65.42% of Go online submissions for Palindrome Number.
//Memory Usage: 6.1 MB, less than 6.19% of Go online submissions for Palindrome Number.
func isPalindrome(x int) bool {
    if x < 0{
		return false
	}
	var inter []int
	for x > 0{
		inter = append(inter, x % 10)
		x = x / 10
	}
	for j := 0; j < len(inter) ; j += 1{
		if inter[j] != inter[len(inter) - j - 1]{
			return false
		}
	}
	return true
}

File: leetcode/3-Palindrome-Number/palindrome-number.cs
#Runtime: 88 ms, faster than 72.86% of C# online submissions for Palindrome Number.
#Memory Usage: 16.9 MB, less than 19.78% of C# online submissions for Palindrome Number.
public class Solution {
    public bool IsPalindrome(int x) {
        if (x < 0){
            return false;
        }
        ArrayList inter = new ArrayList();
        while(x > 0){
            inter.Add(x % 10);
            x = x / 10;
        }
        for(int i = 0; i < inter.Count/2; i++){
            if (!inter[i].Equals(inter[inter.Count - i - 1])){
                return false;
            }
        }
        return true;
    }
}